File: covid.py
#! /usr/bin/env python3

import pandas as pd
import matplotlib.pyplot as plt
covid_state = pd.read_csv("~/DataRepo/covid-19-data/us-states.csv", parse_dates=['date'])
is_IN = covid_state['state'] == 'Indiana'
covid_IN = covid_state[is_IN]
covid_IN = covid_IN.drop(['state','fips'],1)
covid_IN = covid_IN.rename(columns = {"cases" : "IN_Cases", "deaths" : "IN_Deaths"})

covid_county = pd.read_csv("~/DataRepo/covid-19-data/us-counties.csv", parse_dates=['date'])
is_IN = covid_county['state'] == 'Indiana'
is_Mad = covid_county['county'] == 'Madison'
covid_Mad = covid_county[is_IN & is_Mad]
covid_Mad = covid_Mad.drop(['state','county','fips'],1)
covid_Mad = covid_Mad.rename(columns = {"cases" : "Mad_Cases", "deaths" : "Mad_Deaths"})


covid = covid_IN.join(covid_Mad.set_index('date'), on='date')
covid['Mad_Cases'] = covid['Mad_Cases'].fillna(0).astype(int)
covid['Mad_Deaths'] = covid['Mad_Deaths'].fillna(0).astype(int)

thedate = covid['date']
incases = covid['IN_Cases']
fig, ax = plt.subplots()
ax.plot(thedate,incases, color='blue')
ax.set_ylabel('Indiana Cases', color='blue', fontsize=14)

ax2 = ax.twinx()
ax2.plot(thedate,covid['Mad_Cases'], color='green')
ax2.set_ylabel('Madison County Cases', color='green', fontsize=14)
ax.set_title('Indiana COVID-19 Cases and Deaths')
plt.xticks(rotation=45, horizontalalignment='right')
plt.show()

File: covid.R
library(dplyr)
library(ggplot2)

# Pull data from csv files
covid.all <- read.csv("~/DataRepo/covid-19-data/us-states.csv", stringsAsFactors = FALSE)
covid.county <- read.csv("~/DataRepo/covid-19-data/us-counties.csv", stringsAsFactors = FALSE)

# Clean state data
covid.in <-filter(covid.all,state == 'Indiana') %>%
  mutate(date = as.Date(date)) %>%
  rename(IN.Cases = cases, IN.Deaths = deaths) %>%
  select(date,IN.Cases,IN.Deaths)

# Clean county data
covid.madison <-filter(covid.county, county == 'Madison', state == 'Indiana') %>%
  mutate(date = as.Date(date)) %>%
  rename(Mad.Cases = cases, Mad.Deaths = deaths) %>%
  select(date,Mad.Cases,Mad.Deaths)

# Merge state and county data; replace NA with 0
covid <- full_join(covid.in,covid.madison, by="date") %>%
  replace(is.na(.),1)

# Plot State Cases and Deaths
ggplot(covid, aes(x=date)) +
  geom_line(aes(y=IN.Cases, color="Cases"), size=1, alpha=0.9, linetype=1) +
  geom_line(aes(y=IN.Deaths,color="Deaths"), size=1, alpha=0.9, linetype=1) +
  scale_color_manual(name='', values=c('Cases' = 'blue', 'Deaths' = 'red')) +
  scale_x_date(date_breaks = "weeks", date_labels = "%b %d") +
  ggtitle("Indiana COVID-19 Cases and Deaths") +
  xlab("Date") +
  ylab("Count")

# Plot County Cases and Deaths
ggplot(covid, aes(x=date)) +
  geom_line(aes(y=Mad.Cases, color="Cases"), size=1, alpha=0.9, linetype=1) +
  geom_line(aes(y=Mad.Deaths, color="Deaths"), size=1, alpha=0.9, linetype=1) +
  scale_color_manual(name='', values=c('Cases' = 'blue', 'Deaths' = 'red')) +
  scale_x_date(date_breaks = "weeks", date_labels = "%b %d") +
  ggtitle("Madison County COVID-19 Cases and Deaths") +
  xlab("Date") +
  ylab("Count")

ggplot(covid, aes(x=date)) +
  geom_line(aes(y=IN.Cases, color="State"), size=1, alpha=0.9, linetype=1) +
  geom_line(aes(y=Mad.Cases, color="County"), size=1, alpha=0.9, linetype=1) +
  scale_color_manual(name='', values=c('State' = 'blue', 'County' = 'green')) +
  scale_x_date(date_breaks = "weeks", date_labels = "%b %d") +
  ggtitle("State vs County Cases") +
  xlab("Date") +
  ylab("Count")

ggplot(covid, aes(x=date)) +
  geom_line(aes(y=IN.Deaths, color="State"), size=1, alpha=0.9, linetype=1) +
  geom_line(aes(y=Mad.Deaths, color="County"), size=1, alpha=0.9, linetype=1) +
  scale_color_manual(name='', values=c('State' = 'blue', 'County' = 'green')) +
  scale_x_date(date_breaks = "weeks", date_labels = "%b %d") +
  ggtitle("State vs County Deaths") +
  xlab("Date") +
  ylab("Count")